crates/transaction-pool/src/maintain.rs

```diff
@@ -230,6 +241,23 @@ pub async fn maintain_transaction_pool<N, Client, P, St, Tasks>(
             }
         }
 
+        let mut interval_duration = time::interval(config.max_tx_lifetime);
+        tokio::select! {
+            _ = interval_duration.tick() => {
+                let old_tx_hashes: Vec<_> = pool
+                    .queued_transactions()
+                    .into_iter()
+                    .filter(|tx| tx.timestamp.elapsed() > config.max_tx_lifetime)
+                    .filter_map(|tx| Arc::try_unwrap(tx).ok())
+                    .map(|tx| *tx.hash())
+                    .collect();
+
+                if !old_tx_hashes.is_empty() {
+                    pool.remove_transactions(old_tx_hashes);
+                }
```

Comment:
this arm we need to integrate into the existing select! macro:

https://github.com/paradigmxyz/reth/blob/06132f509cdc498d01ccbec9eed4fbbf85bd1e0e/crates/transaction-pool/src/maintain.rs#L217-L217